### PR TITLE
Reorganize the fields of the global context.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by zherczeg** [comment]
The size of globals is reduced by 4 bytes. The effect is just a little noise.

**Comment by LaszloLango** [comment]
Not a big gain :(
LGTM

File: jerry-core/jcontext/jcontext.h [hunk 1]
```diff
@@ -47,58 +46,45 @@
...
+  uint8_t is_direct_eval_form_call; /**< direct call from eval */
+  uint8_t jerry_api_available; /**< API availability flag */
+#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
+  uint8_t re_cache_idx; /**< evicted item index when regex cache is full (round-robin) */
+#endif /* !CONFIG_DISABLE_REGEXP_BUILTIN */
 
 #ifdef JMEM_STATS
   jmem_heap_stats_t jmem_heap_stats; /**< heap's memory usage statistics */
   jmem_pools_stats_t jmem_pools_stats; /**< pools' memory usage statistics */
 #endif /* MEM_STATS */
```
Comment (akosthekiss) [inline jerry-core/jcontext/jcontext.h]: Since already touching everything here, could you also fix the comment here (prepend a J to MEM_STATS)?

**Comment by akosthekiss** [comment]
After the comment fix, LGTM

**Comment by zherczeg** [comment]
Thank you

**Pull request merged**